decookies/wasm-vm/src/types.jsligo

```diff
@@ -32,15 +33,29 @@ type actions =
   | ["Bank"]
   | ["Temple"];
 
-type operationType = | ["Mint"] | ["Transfer"] | ["Eat"];
+type operationType = | ["Mint"] | ["Transfer"] | ["Eat"] | ["Delegate"];
 
-type parameter =
+type mint =
   {
-    operationType,
-    operation: option<actions>,
+    operation: actions,
     amount: nat,
-    recipient: option<address>
+    layerOneAddress: address
   };
+type transfer =
+  {
+    amount: nat,
+    recipient: address,
+    layerOneAddress: address
+  };
+type eat =
+  {
+    amount: nat,
+    layerOneAddress: address
+  };
+
+type delegate = {layerOneAddress: address, address: address};
```

Comment:
I don't think this operation should have a second `address` field. I think this should come directly from `Tezos.get_sender ()`.